analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.FlowCaptureOperation.cs

```diff
@@ -0,0 +1,354 @@
+            """;
+        var context = CreateContextCS(code, additionalParameters: "string st, string st2");
+        context.ValidateEntry(LiveIn("boolParameter", "st", "st2"), LiveOut("boolParameter", "st", "st2"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("boolParameter", "st", "st2"), LiveOut("boolParameter", "st", "st2"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("boolParameter", "st", "st2"), LiveOut("boolParameter", "st", "st2"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("boolParameter", "st", "st2"), LiveOut("boolParameter", "st", "st2"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("boolParameter", "st", "st2"), LiveOut("boolParameter", "st2"));
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("boolParameter", "st2"), LiveOut("st2"));
+        context.Validate(context.Cfg.Blocks[6], null, LiveIn("st2"), LiveOut("st2"));
+        context.Validate(context.Cfg.Blocks[7], null, LiveIn("st2"), LiveOut("st2"));
+        context.Validate(context.Cfg.Blocks[8], null, LiveIn("st2"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingOperator_ConsequentCalls()
+    {
+        /*       Block 1
+         *       #0=s1
+         *     #0 is null
+         *        /   \
+         *       F     T
+         *       /     \
+         *    Block 2  Block 3 -+---F--> Block 4 --------+
+         *    #1=#0    #2=s2    |       #1=#2            |
+         *       \  #2 is null  |                        |
+         *        \     /       |                        |
+         *         Block 6 <-+  +--T--> Block 5--------->|
+         *          s=#1     |         #1="Hello"        |
+         *           |       +---------------------------+
+         *         Exit
+         */
+        const string code = """string s = s1 ?? s2 ?? "Hello";""";
+        var context = CreateContextCS(code, additionalParameters: "string s1, string s2");
+        context.ValidateEntry(LiveIn("s1", "s2"), LiveOut("s1", "s2"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("s1", "s2"), LiveOut("s1", "s2"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("s1", "s2"), LiveOut("s1", "s2"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("s1", "s2"), LiveOut("s1", "s2"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("s1", "s2"), LiveOut("s1", "s2"));
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("s1", "s2"), LiveOut("s1", "s2"));
+        context.Validate(context.Cfg.Blocks[6], null, LiveIn("s1", "s2"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingOperator_ConsequentCalls_Assignment()
+    {
+        const string code = """string s = s1 ??= s2 = s3 ??= s4 ?? "End";""";
+        var context = CreateContextCS(code, additionalParameters: "string s1, string s2, string s3, string s4");
```

Comment:
These comments do not help much. Write a semantic meaning of the block instead, like in `FlowCaptrure_SwitchStatement`